### ISSUE TITLE
Improving exception handling

### ISSUE THREAD

**Issue by andrei-shabanski** [spothero/py-responsys#13]
Now, you're throwing ResponsysClientError when the server returns an error. That gives the opportunity to catch server's errors and handle them. But ResponsysClientError doesn't provide information about what happened wrong. The error message with server's response is the only thing we can get.

How do you look at storing server's response (`errorCode`, `title`, `detail`, `errorDetails`) in ResponsysClientError?

**Comment by nbkincaid** [comment]
@andrei-shabanski Yes, this is definitely crucial. I'm sorry for being away from this for so long. I will get in a pull request for this.

**Comment by nbkincaid** [comment]
@andrei-shabanski Actually it looks like you already have most of this finished already here: https://github.com/andrei-shabanski/py-responsys/commit/90e7f2098c2ae67c34dbb5d9dbe1f046bb05b168

If you open up a pr, I will go ahead and merge it in.
